mesonbuild/cmake/traceparser.py

```diff
@@ -672,11 +672,11 @@ def _parse_common_target_options(self, func: str, private_prop: str, interface_p
             if i in ignore:
                 continue
 
-            if i in {'INTERFACE', 'LINK_INTERFACE_LIBRARIES', 'PUBLIC', 'PRIVATE', 'LINK_PUBLIC', 'LINK_PRIVATE'}:
+            if i in {'INTERFACE', 'INTERFACE_LINK_LIBRARIES', 'PUBLIC', 'PRIVATE', 'LINK_PUBLIC', 'LINK_PRIVATE'}:
```

Comment:
I think this is a situation where we should look at both. [LINK_INTERFACE_LIBRARIES](https://cmake.org/cmake/help/latest/prop_tgt/LINK_INTERFACE_LIBRARIES.html) is deprecated and replaced by INTERFACE_LINK_LIBRARIES. LINK_INTERFACE_LIBRARIES will be ignored if INTERFACE_LINK_LIBRARIES is also given. Which leads to the interesting question of what does the trace parser do if you pass both.

Reply:
If both are added here, it will merge them - which is not `CMake` would do - but this is a broken file anyway.

Reply:
It will take whatever the CMake trace parser spits out. I'm hoping that the trace parser will do the right thing and ignore the LINK_INTERFACE_LIBRARIES if INTERFACE_LINK_LIBRARIES is also set, but if it doesn't then we need to handle that.

Reply:
For now though, I'm happy to just add the extra value and hope for the best.